reana_client/cli/workflow.py

```diff
@@ -738,6 +741,18 @@ def workflow_logs(ctx, workflow, access_token, json_format, steps=None):  # noqa
                 steps = list(set(steps))
             response = get_workflow_logs(workflow, access_token, steps)
             workflow_logs = json.loads(response["logs"])
+
+            if failed:
+                steps = list(
+                    workflow_logs["job_logs"][item]["job_name"]
```

Comment:
I think you can simplify to `[key for key, value in workflow_logs["job_logs"].items() if value["status"] == "failed"]`

Reply:
I will try approach two :)